src/server/aws_lambda.ts

```diff
@@ -49,21 +53,40 @@ function syncHandler<TEvent = any, TResult = any>(handler: SyncHandler<TEvent, T
     return function wrappedLambdaHandler(event, context, cb) {
         hb.__setStore(AsyncStore)
         AsyncStore.run({context: {}, breadcrumbs: []}, () => {
+            const timeoutHandler = setupTimeoutWarning(hb, context)
             try {
                 handler(event, context, (error, result) => {
+                    clearTimeout(timeoutHandler)
                     if (error) {
                         return reportToHoneybadger(hb, error, cb)
                     }
 
                     cb(null, result)
                 });
             } catch (err) {
+                clearTimeout(timeoutHandler)
                 reportToHoneybadger(hb, err, cb)
             }
         })
     }
 }
 
+function shouldReportTimeoutWarning(hb: Honeybadger, context: Context): boolean {
+    return typeof context.getRemainingTimeInMillis === 'function' && !!((hb.config as ServerlessConfig).reportTimeoutWarning)
+}
+
+function setupTimeoutWarning(hb: Honeybadger, context: Context) {
+    if (!shouldReportTimeoutWarning(hb, context)) {
+        return
+    }
+
+    const delay = context.getRemainingTimeInMillis() - ((hb.config as ServerlessConfig).timeoutWarningThresholdMs)
```

Comment:
`delay`'s a bit of a confusing name for me here. Maybe `timeLeftToMarkAsTimeout`? A bit wordy, I know, but I wasn't sure what was being delayed at first (which kinda ties into my other point of decoupling the two.)

Reply:
I name it `delay` because that's how much we _delay_ the execution of the callback that notifies Honeybadger. I have no objection with longer variable names 😄  ; if you still think it's confusing, we can change it.